### PR TITLE
{devel}[gcccuda/2020b] hipify-clang 4.2.0

### PR DESCRIPTION
**Pull request by nordmoen**
`hipify-clang` is a clang-based tool for translating CUDA sources into HIP sources.

### CODE REVIEW

File: easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    # ('Python', '3.8.6'),
+]
+
+configopts = '-DHIPIFY_CLANG_TESTS=0'
+configopts += ' -DCMAKE_INSTALL_PREFIX=%(installdir)s'
+configopts += ' -DCMAKE_PREFIX_PATH=$EBROOTCLANG'
+# NOTE: The following configuration options are needed if building with tests
+# (as activated by 'runtest' below), for these to work the above commented
+# dependencies need to be uncommented. Currently testing is disabled due to a
+# lack of 'llvm-lit'.
```
Comment (boegel) [inline easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb:35]: What's `llvm-lit` exactly, do you have a link for that?

Comment (nordmoen) [reply easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb:35]: I have not looked too much into this, but it is apparently [LLVM's own test framework](https://www.llvm.org/docs/CommandGuide/lit.html). It is built as a part of LLVM [when testing is enabled](https://github.com/llvm/llvm-project/blob/main/llvm/CMakeLists.txt#L1000).

File: easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+]
+
+dependencies = [
+    # ('cuDNN', '8.0.4.30', '-CUDA-11.1.1', True),
+    ('Clang', '11.0.1'),
+    # ('Python', '3.8.6'),
+]
+
+configopts = '-DHIPIFY_CLANG_TESTS=0'
+configopts += ' -DCMAKE_INSTALL_PREFIX=%(installdir)s'
```
Comment (boegel) [inline easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb:30]: This is added automatically by `CMakeMake`, should not be needed?

File: easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+builddependencies = [
+    ('CMake', '3.18.4'),
+    ('zlib', '1.2.11'),
+]
+
+dependencies = [
+    # ('cuDNN', '8.0.4.30', '-CUDA-11.1.1', True),
+    ('Clang', '11.0.1'),
+    # ('Python', '3.8.6'),
```
Comment (boegel) [inline easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb:26]: I would keep the commented out dependencies together, and maybe even comment them out but without running the tests until we have `llvm-lit` available as a build dep?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# (as activated by 'runtest' below), for these to work the above commented
+# dependencies need to be uncommented. Currently testing is disabled due to a
+# lack of 'llvm-lit'.
+# configopts += ' -DCUDA_TOOLKIT_ROOT_DIR=$EBROOTCUDA'
+# configopts += ' -DCUDA_DNN_ROOT_DIR=$EBROOTCUDNN'
+# configopts += ' -DPYTHON_EXECUTABLE=$EBROOTPYTHON/bin/python'
+
+# runtest = 'test-hipify'
+
+# For some reason the executable is not placed in 'bin'
```
Comment (boegel) [inline easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb:42]: That must be a bug, no? Should we report this upstream?

Perhaps @Flamefire has some feedback here on how to fix this properly in the `CMakeLists.txt`...

File: easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+dependencies = [
+    # ('cuDNN', '8.0.4.30', '-CUDA-11.1.1', True),
+    ('Clang', '11.0.1'),
+    # ('Python', '3.8.6'),
+]
+
+configopts = '-DHIPIFY_CLANG_TESTS=0'
+configopts += ' -DCMAKE_INSTALL_PREFIX=%(installdir)s'
+configopts += ' -DCMAKE_PREFIX_PATH=$EBROOTCLANG'
```
Comment (Flamefire) [inline easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb:31]: Not required, set automatically and is even potentially harmful: We have set this as an env var and I'd need to look up if this overwrites (and hence removes) our stuff or not.
Anyway, this is already set so no need to do here :)

File: easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb (hunk 4 above)
Comment (Flamefire) [reply easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb:42]: Reported as https://github.com/ROCm-Developer-Tools/HIPIFY/issues/340

To much work already to fix that, sorry. Essentially you would want to replace `HIPIFY_INSTALL_PATH` at https://github.com/ROCm-Developer-Tools/HIPIFY/blob/397b58a8fc38fa9c5c3252ccfa6e9cd1f0f141a5/CMakeLists.txt#L105 by `bin` and at https://github.com/ROCm-Developer-Tools/HIPIFY/blob/397b58a8fc38fa9c5c3252ccfa6e9cd1f0f141a5/CMakeLists.txt#L109 by `.` but doing the properly with their setting-stuff and HIP integration is more work. So I'm fine with this postinstall step

**Comment by nordmoen** [comment]
Made the requested changes

File: easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    ('zlib', '1.2.11'),
+]
+
+dependencies = [
+    ('Clang', '11.0.1'),
+    # ('cuDNN', '8.0.4.30', '-CUDA-11.1.1', True),
+    # ('Python', '3.8.6'),
+]
+
+configopts = '-DHIPIFY_CLANG_TESTS=0'
```
Comment (Flamefire) [inline easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb:29]: As this is off by default, I think it makes sense to set it to 1 and comment it out next to the other test-related options below.

Comment (nordmoen) [reply easybuild/easyconfigs/h/hipify-clang/hipify-clang-4.2.0-gcccuda-2020b.eb]: Done :+1:

**Review by Flamefire — APPROVED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2617.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/e52ba36ead7426efe9c4d3f62fbc7949 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3503.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/32b374af9d2cc2903798dc0925802f39 for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @nordmoen!

**Pull request merged**